src/main/java/com/aerohockey/mechanics/MechanicsExecutor.java

```diff
@@ -1,55 +1,132 @@
 import java.util.concurrent.Executors;
 
 /**
  * Created by sergeybutorin on 16.04.17.
  */
 @Service
-public class MechanicsExecutor implements Runnable {
+public class MechanicsExecutor {
     private static final long STEP_TIME = 30;
 
-    private final @NotNull GameMechanics gameMechanics;
+    private static final int THREADS = 4;
 
-    private final @NotNull Clock clock = Clock.systemDefaultZone();
+    private final AccountService accountService;
 
-    private final Executor tickExecutor = Executors.newSingleThreadExecutor();
+    private final ServerSnapService serverSnapService;
+
+    private final ServerDetailSnapService serverDetailSnapService;
+
+    private final RemotePointService remotePointService;
+
+    private final GameOverSnapService gameOverSnapService;
+
+    private final ExecutorService tickExecutors = Executors.newFixedThreadPool(THREADS, Thread::new);
+
+    private final GameMechanics[] gameMechanics = new GameMechanics[THREADS];
 
     @Autowired
-    public MechanicsExecutor(@NotNull GameMechanics gameMechanics) {
-        this.gameMechanics = gameMechanics;
+    public MechanicsExecutor(AccountService accountService, ServerSnapService serverSnapService, ServerDetailSnapService serverDetailSnapService, RemotePointService remotePointService, BallMovementService ballMovementService, GameOverSnapService gameOverSnapService) {
+        this.accountService = accountService;
+        this.serverSnapService = serverSnapService;
+        this.serverDetailSnapService = serverDetailSnapService;
+        this.remotePointService = remotePointService;
+        this.gameOverSnapService = gameOverSnapService;
     }
 
     @PostConstruct
     public void initAfterStartup() {
-        tickExecutor.execute(this);
+        for (int i = 0; i < gameMechanics.length; i++) {
+            gameMechanics[i] = new GameMechanicsImpl(accountService, serverSnapService,
+                    remotePointService,
+                    gameOverSnapService,
+                    serverDetailSnapService);
+            final Runnable mechanicsRunnable = new MechanicsRunnable(gameMechanics[i]);
+            tickExecutors.execute(mechanicsRunnable);
+        }
     }
 
-    @Override
-    public void run() {
-        long lastFrameMillis = STEP_TIME;
-        while (true) {
-            final long before = clock.millis();
-
-            gameMechanics.gmStep(lastFrameMillis);
+    public void addUser(@NotNull Long userId) {
+        for (GameMechanics mechanics : gameMechanics) {
+            if (mechanics.isUserWaiting(userId) || mechanics.isUserPlaying(userId)) {
+                return;
+            } else if (mechanics.isCandidatesExists()) {
+                mechanics.addUser(userId);
+                return;
+            }
+        }
 
-            final long after = clock.millis();
-            TimeHelper.sleep(STEP_TIME - (after - before));
+        GameMechanics selectedMechanics = gameMechanics[0];
```

Comment:
`Optional<GameMechanics> mechanics = Stream.of(gameMechanics).min(Comparator.comparingInt(GameMechanics::getSessionCount));`